client/db/bolt/upgrades.go

```diff
@@ -0,0 +1,91 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package bolt
+
+import (
+	"fmt"
+
+	"decred.org/dcrdex/dex/encode"
+	"go.etcd.io/bbolt"
+)
+
+const (
+	initialVersion = 0
+
+	// DBVersion is the latest version of the database that is understood by the
+	// program. Databases with recorded versions higher than this will fail to
+	// open (meaning any upgrades prevent reverting to older software).
+	DBVersion = initialVersion
+)
+
+// upgrade the database to the next version. Each database upgrade function
+// should be keyed by the database version it upgrades. For example, lets say
+// we have a new version which upgrades match IDs. Its associated upgrade
+// entry would look like this:
+// 			matchIDVersion - 1: matchIDUpgrade
+//
+var upgrades = [...]func(tx *bbolt.Tx) error{}
```

Comment:
It's unfortunate that we have no way to test `upgradeDB` properly with an empty but constant length array.

Reply:
I'm going to make the make persisting the version the first upgrade here, would resolve the lack of tests issue and also provide a working example to reference for future upgrades.